PWGCF/Femto3D/Tasks/femto3dPairTask.cxx

```diff
@@ -0,0 +1,305 @@
+
+        for (auto indx : comb_idx) {
+          FemtoPair* Pair = new FemtoPair((i->second)[indx[0]], (i->second)[indx[1]], IsIdentical);
+          if (!Pair->IsClosePair(_deta, _dphi, _radiusTPC))
+            registry.fill(HIST("SE"), Pair->GetKstar()); // close pair separation doesn't work properly, need eta and phi at some position in the barrel
+        }
+      } // end of same event identical
+
+      for (auto i = mixbins.begin(); i != mixbins.end(); i++) { // start of mixed event identical
+        if ((i->second).size() < 2)
+          continue;
+
+        std::vector<std::vector<int>> comb_idx;
+        comb(comb_idx, (i->second).size(), 2);
+
+        for (auto indx : comb_idx) {
+          int colId_1 = (i->second)[indx[0]];
+          int colId_2 = (i->second)[indx[1]];
+
+          for (auto ii : selectedtracks_1[colId_1]) {
+            for (auto iii : selectedtracks_1[colId_2]) {
+              FemtoPair* Pair = new FemtoPair(ii, iii, IsIdentical);
+              if (!Pair->IsClosePair(_deta, _dphi, _radiusTPC))
+                registry.fill(HIST("ME"), Pair->GetKstar());
+            }
+          }
+        }
+      } // end of mixed event identical
+
+    } //====================================== end of mixing identical ======================================
+
+    else { //====================================== mixing non-identical ======================================
+
+      for (auto i = selectedtracks_1.begin(); i != selectedtracks_1.end(); i++) { // start of same event non-identical
+
+        auto j = selectedtracks_2.find(i->first);
+        if (j == selectedtracks_2.end())
+          continue;
+
+        for (auto ii : i->second) {
+          for (auto iii : j->second) {
+            FemtoPair* Pair = new FemtoPair(ii, iii, IsIdentical);
+            if (!Pair->IsClosePair(_deta, _dphi, _radiusTPC))
+              registry.fill(HIST("SE"), Pair->GetKstar()); // close pair separation doesn't work properly, need eta and phi at some position in the barrel
+          }
+        }
+      } // end of same event non-identical
+
+      for (auto i = mixbins.begin(); i != mixbins.end(); i++) { // start of mixed event non-identical
+        if ((i->second).size() < 2)
+          continue;
+
+        std::vector<std::vector<int>> comb_idx;
+        comb(comb_idx, (i->second).size(), 2);
+
+        for (auto indx : comb_idx) {
+          int colId_1 = (i->second)[indx[0]];
+          int colId_2 = (i->second)[indx[1]];
+
+          for (auto ii : selectedtracks_1[colId_1]) {
+            for (auto iii : selectedtracks_2[colId_2]) {
+              FemtoPair* Pair = new FemtoPair(ii, iii, IsIdentical);
+              if (!Pair->IsClosePair(_deta, _dphi, _radiusTPC))
+                registry.fill(HIST("ME"), Pair->GetKstar());
+            }
+          }
+        }
+      } // end of mixed event non-idetical
+
+    } //====================================== end of mixing non-identical ======================================
```

Comment:
Same comment as above

Reply:
Here you're right. I fixed this and now I use only one "FemtoPair" object and just change the stored objects inside it. Thanks for pointing it out to me.